### PR TITLE
add : auth UI + backend

### LINKED ISSUE #1: Sign up and log-in for students
- [x] Sign up page should take Name ( first and last); email, create password and "tnc" checkbox 
- [x] Sign up page should take first and last name in different fields 
- [x] Sign up page should take email address of the signee
- [x] Sign up page should ask the signee to create the password : Create password field should have two conditions: a. to make have one or 2 numbers b. should be more than 8 characters. 
- [x] Password input field should also have an eye icon with the functionality to view the password otherwise the password will be hidden..
- [x] Sign up page should have a check box for agreeing to terms and policies, which must be checked before creating the account. 
- [x] Clicking on the "Terms" and "privacy Policy" will take the students/signee to the handbook page of terms and policies.
- [x] Sign up page should have "Create Account" button: This button shall be enabled only after the data in above mentioned fields are filled by the signee and the check .
- [x] Create Account button: This button will store all the data taken in above mentioned input fields and store it in the. main database
- [x] Sign up page should have a "Log-in instead" option for students who already have signed up : this option will take them to the Log-in Page.
-[ ] Figma prototype: 
image

LOG IN PAGE

- [x] The "log-in instead" button should take the student to the log-in page.
- [x] The log-in page should take only email and password 
- [x] The Password input field should also have an eye icon with the functionality to view the password the student is typing in. Otherwise the password will be hidden.
- [x] The password field should have a "forgot password" option. Clicking which will enable the student to create new password or retrieve their old password.
- [x] The log-in page should have a log-in button:  which will be enabled only when the provided data in the email and password fields are checked up & matched with the sign-up details that was stored. 
- [x] In the event the email or password doesn't match or the data provided isn't as per any stored signee data, a message indicating the same should be shown to the student.
- [x] The log-in button should take the student to its "student dashboard"
- [x] The log-in page should also have a "create account" option which shall take the user to the "sign up " page.
- [x] The prototype is below.
image

Figma file to refer: https://www.figma.com/file/AvkiFXfsMSEOQFp68p1g6N/ADM-System?node-id=19%3A289
PRD : https://www.notion.so/Product-Spec-Platform-3-4-5f2aaaf2b98b425b89923f9a2b48e649
Wiring : https://www.figma.com/proto/AvkiFXfsMSEOQFp68p1g6N/ADM-System?node-id=80%3A407&scaling=min-zoom&page-id=0%3A1

### PR DESCRIPTION
**Pull request by OmkarK45**
# Closes #1 

### Changes

- Add: auth folder in pages directory
- Add: support for login and signup
- Add: AuthLayout component

### Screenshots or Video GIFs

Login page 
image

Register Page with all validation messages
image

image

### Related Issues
- Issue #1

### CODE REVIEW

File: client/src/components/Layout/AuthLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  children: React.ReactNode
+}
+export function AuthLayout({ children }: Props) {
+  return (
+    <Stack
+      minH="calc(100vh - 60px);"
+      d="flex"
+      align={['', '', 'center']}
+      justify="center"
+      bg={'#151515'}
```
Comment (MehulKChaudhari) [inline client/src/components/Layout/AuthLayout.tsx:14]: here you can use `theme.colors.black["500"]` instead `#151515`

File: client/src/components/Layout/AuthLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    >
+      <Stack
+        maxW={['100%', '100%', '80%']}
+        direction={{ base: 'column', md: 'row' }}
+        rounded={'10px'}
+        overflow="hidden"
+        bg={'black.800'}
```
Comment (MehulKChaudhari) [inline client/src/components/Layout/AuthLayout.tsx:21]: here same use theme for adding colors

File: client/src/context/AuthContext.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  authState: defaultAuthState,
+  setState: () => {},
+  logoutUser: () => {},
+})
+
+export function AuthProvider({ children }: { children: ReactNode }) {
+  const [authState, setAuthState] = useState<IAuthState>(defaultAuthState)
+
+  useEffect(() => {
+    const getUserInfo = async () => {
```
Comment (uditkumar01) [inline client/src/context/AuthContext.tsx:48]: remove .then from async function

File: client/src/pages/auth/login.tsx [hunk 4]
```diff
@@ -1,3 +1,193 @@
...
+})
+
 export default function Login() {
-  return <div>Login Page</div>
+  const { setState } = useAuth()
+  const router = useRouter()
+  const toast = useToast()
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+
+  async function handleSubmit(data: LoginValues) {
```
Comment (uditkumar01) [inline client/src/pages/auth/login.tsx:42]: export this function into a new file

File: client/src/pages/auth/login.tsx [hunk 5]
```diff
@@ -1,3 +1,193 @@
...
+          title: 'Failed to log you in.',
+          description: error.response.data.msg,
+          status: 'error',
+          isClosable: true,
+        })
+      })
+  }
+
+  return (
+    <>
```
Comment (uditkumar01) [inline client/src/pages/auth/login.tsx:83]: make reuseable components

**Review by uditkumar01 — COMMENTED** [review]
.Plz, try to fix these issues.

**Review by MehulKChaudhari — CHANGES_REQUESTED** [review]
Great awesome, I have added some changes please look into it.

File: client/src/context/AuthContext.tsx (hunk 3 above)
Comment (OmkarK45) [reply client/src/context/AuthContext.tsx:48]: Noted. I will fix it for sure.

**Pull request merged**